.storybook/lucid-docs-addon/index.js

```diff
@@ -0,0 +1,139 @@
+	return _.isUndefined(defaultValue)
+		? _.get(componentRef, ['peekDefaultProps', property])
+		: defaultValue;
+};
+
+const getPropTypeData = resolverFn => {
+	const type = _.get(resolverFn, ['peek', 'type']);
+
+	let oneOfData;
+	if (type === 'oneOf') {
+		oneOfData = _.get(resolverFn, ['peek', 'args', 0]);
+	}
+
+	let arrayOfData;
+	if (type === 'arrayOf') {
+		const arrayOfResolverFn = _.get(resolverFn, ['peek', 'args', 0]);
+		arrayOfData = getPropTypeData(arrayOfResolverFn);
+	}
+
+	let oneOfTypeData;
+	if (type === 'oneOfType') {
+		const oneOfTypeResolverFns = _.get(resolverFn, ['peek', 'args', 0]);
+		oneOfTypeData = _.map(oneOfTypeResolverFns, getPropTypeData);
+	}
+
+	let instanceOfData;
+	if (type === 'instanceOf') {
+		const instanceOfClass = _.get(resolverFn, ['peek', 'args', 0]);
+		instanceOfData = instanceOfClass.name;
+	}
+
+	let objectOfData;
+	if (type === 'objectOf') {
+		const objectOfResolverFn = _.get(resolverFn, ['peek', 'args', 0]);
+		objectOfData = getPropTypeData(objectOfResolverFn);
+	}
+
+	let shapeData;
+	if (type === 'shape') {
+		const shapeObject = _.get(resolverFn, ['peek', 'args', 0]);
+		shapeData = _.mapValues(shapeObject, getPropTypeData);
+	}
+
+	return {
+		...resolverFn.peek,
+		type,
+		oneOfData,
+		arrayOfData,
+		oneOfTypeData,
+		instanceOfData,
+		objectOfData,
+		shapeData,
+	};
+};
+
+const getPropsData = componentRef => {
+	return _.map(componentRef.propTypes, (resolverFn, property) => {
+		return {
+			...getPropTypeData(resolverFn),
+			name: property,
+			defaultValue: getDefaultPropValue(componentRef, property),
+		};
+	});
+};
+
+export const withDescription = componentRef => {
+	return StoryComponent => {
+		return ({ kind, story }) => {
+			return (
+				<div>
+					<h1>
+						{kind}
+					</h1>
+					<section>
+						{compile(stripIndent(componentRef.peek.description)).tree}
+					</section>
+					<h2>
+						{story}
+					</h2>
+					<StoryComponent {...{ kind, story }} />
+				</div>
+			);
+		};
+	};
+};
+
+export const withProps = componentRef => {
+	return StoryComponent => {
+		return props => {
+			const channel = addons.getChannel();
+			channel.emit(
+				'lucid-docs-display-props',
+				JSON.stringify(getPropsData(componentRef))
+			);
+			return <StoryComponent {...props} />;
+		};
+	};
+};
+
+export const withCode = (source = null) => {
+	return StoryComponent => {
+		return props => {
+			const channel = addons.getChannel();
+			channel.emit('lucid-docs-source', source);
+
+			return <StoryComponent {...props} />;
+		};
+	};
+};
+
+const getDefaultExport = mod => (mod.__esModule ? mod.default : mod);
+
+export const exampleStory = ({ component, code, example, options }) => {
+	let StoryComponent = getDefaultExport(example);
+	if (options) {
+		StoryComponent = props => {
+			setOptions(options);
+			return React.createElement(getDefaultExport(example), props);
+		};
+	}
+
+	const componentRef = getDefaultExport(component);
+
+	return withDescription(componentRef)(
```

Comment:
this function call is kinda complex, breaking it out into named variables passed in would maybe make it clearer what's going on